gateway/api/middleware.go

```diff
@@ -26,23 +31,48 @@ import (
 
 var errInvalidAuthHeaderErr = errors.New("invalid authorization header")
 
-func (a *Api) Authenticate(c *gin.Context) {
-	roleName := RoleFromContext(c)
-	subject, err := a.validateAccessToken(c)
+func (a *Api) localAuthMiddleware(c *gin.Context) {
+	tokenString := c.GetHeader("Authorization")
+	// remove bearer from token
+	tokenString = strings.Replace(tokenString, "Bearer ", "", 1)
+	if tokenString == "" {
+		c.JSON(http.StatusUnauthorized, gin.H{"error": "Missing authorization token"})
+		c.Abort()
+		return
+	}
+
+	jwtKey := appconfig.Get().JWTSecretKey()
+	claims := &localauthapi.Claims{}
+	token, err := jwt.ParseWithClaims(tokenString, claims, func(token *jwt.Token) (interface{}, error) {
+		return jwtKey, nil
+	})
+
+	if err != nil || !token.Valid {
+		c.JSON(http.StatusUnauthorized, gin.H{"error": "Invalid or expired token"})
+		c.Abort()
+		return
+	}
+
+	sessionByToken, err := pglocalauthsession.GetSessionByToken(tokenString)
 	if err != nil {
-		tokenHeader := c.GetHeader("authorization")
-		log.Infof("failed authenticating, %v, length=%v, reason=%v",
-			parseHeaderForDebug(tokenHeader), len(tokenHeader), err)
-		c.AbortWithStatus(http.StatusUnauthorized)
+		fmt.Printf("err: %v\n", err)
```

Comment:
```suggestion
		log.Errorf("err: %v\n", err)
```

Also describe better the error

Reply:
this was removed already

Reply:
thanks